src/main/java/org/folio/circulation/domain/RequestStatus.java

```diff
@@ -2,31 +2,38 @@
 
 import static org.apache.commons.lang3.StringUtils.equalsIgnoreCase;
 
-import java.util.ArrayList;
 import java.util.Arrays;
-
-import org.folio.circulation.domain.representations.RequestProperties;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.stream.Collectors;
 
 import io.vertx.core.json.JsonObject;
+import org.apache.commons.lang3.StringUtils;
+
+import org.folio.circulation.domain.representations.RequestProperties;
 
 public enum RequestStatus {
   NONE(""),
   OPEN_NOT_YET_FILLED("Open - Not yet filled"),
   OPEN_AWAITING_PICKUP("Open - Awaiting pickup"),
   OPEN_IN_TRANSIT("Open - In transit"),
+  OPEN_AWAITING_DELIVERY("Open - Awaiting delivery"),
   CLOSED_FILLED("Closed - Filled"),
   CLOSED_CANCELLED("Closed - Cancelled"),
   CLOSED_UNFILLED("Closed - Unfilled"),
   CLOSED_PICKUP_EXPIRED("Closed - Pickup expired");
 
+  private static final EnumSet<RequestStatus> VALID_STATUSES =
+    EnumSet.range(OPEN_NOT_YET_FILLED, CLOSED_PICKUP_EXPIRED);
+  private static final EnumSet<RequestStatus> OPEN_STATUSES =
+    EnumSet.range(OPEN_NOT_YET_FILLED, OPEN_AWAITING_DELIVERY);
+
   private final String value;
 
   public static String invalidStatusErrorMessage() {
-    //TODO: Generalise this to join all states
-    return String.format("Request status must be \"%s\", \"%s\", \"%s\", \"%s\", \"%s\" or \"%s\"",
-      OPEN_NOT_YET_FILLED.getValue(), OPEN_AWAITING_PICKUP.getValue(),
-      OPEN_IN_TRANSIT.getValue(), CLOSED_FILLED.getValue(),
-      CLOSED_UNFILLED.getValue(), CLOSED_PICKUP_EXPIRED.getValue());
+    return "Request status must be one of the following: " +
+      VALID_STATUSES.stream().map(s -> StringUtils.wrap(s.getValue(), '"'))
+        .collect(Collectors.joining(", "));
```

Comment:
Will this include the `NONE` status, probably as "", as a possible status choice? I see the tests were changed to just check the message preamble, which is why I ask.

Reply:
Oops, never mind, I see you created an enum set to handle this.

Reply:
Though, creating a subset seems pretty heavy-weight just to deal with a, one would hope, relatively rare error condition. Also using a range implies order to the enum. I don't think there should be any expectation of order for a string based status enumeration. If we must have an enum set, could we be explicit in construction so future statuses don't cause unexpected behavior if placed at random spots in the enum? I think being explicit in the creation of the enum set is better than assuming order and using a range.

Reply:
That is my primary concern with the enum set. What happens if a new enumeration is added, or the existing ones reordered? Does that change the definition of the set?

Especially as there doesn’t appear to be explicit tests for the behaviour backed by these sets.